qiskit_experiments/framework/backend_data.py

```diff
@@ -0,0 +1,145 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Backend data access helper class
+
+Since `BackendV1` and `BackendV2` do not share the same interface, this
+class unifies data access for various data fields.
+"""
+from qiskit.providers import BackendV1, BackendV2
+
+
+class BackendData:
```

Comment:
This is just using a class as a namespace, so I don't see any fundamental difference from defining a set of functions. I would turn this into an actual class with constructor taking a backend, so that you can store this backend configuration as an experiment or run options, i.e. note that currently we cannot serialize backend and thus hard-coded run/transpile options in a backend object are all discarded in the loaded experiment.

Perhaps it's useful to override the base experiment `_set_backend` method to internally generate this data and set the data instance to the experiment instance. Then all experiment subclasses can easily access these configuration data, e.g.

```python
self._backend_data.granularity
```

Reply:
I like the suggestion of making this a full class that takes a backend as a constructor argument and then making most of the methods into properties. Setting it up from `_set_backend` makes sense as well.

Making it serializable is also nice, though I don't know if it is worth it to serialize and attach the backend data to each experiment.

Reply:
For now we don't attach anything to experiment instance IIRC. These will be separately saved as an artifact.

Reply:
I agree, I'll do it as the last touch on the PR, in case we get more design ideas by then.

Reply:
ok, done.